### DISCUSSION TITLE
Tiddlywiki.com stylesheets AND highlight plugin

### DISCUSSION THREAD

**Discussion by AnthonyMuscio** [TiddlyWiki/TiddlyWiki5#6838] [Ideas]
I was thinking given tiddlywiki.com documents various things including stylesheets perhaps it should have the highlight plugin installed just as it has the internals plugin. 

I did install the highlight plugin but then was surprised to discover, some at least of the items with the tag $:/tags/Stylesheets in fact have the tiddler type empty or text rather than set to text/css.
* We can not use a tiddler type of "CSS" in some cases because the style sheet includes macro definitions, and thus any css within it not highlighted.

**I just wanted to raise the suggestion that we;**
* Install the highlight plugin on tiddlywiki.com
* set the tiddlers with the tag $:/tags/Stylesheets to the correct tiddler type (not containing macros)
* Determine if there is a way to highlight the css in a tiddler also containing macros without breaking the css.
  * my guess is allowing the css highlighter to also operate on a code view of stylesheets containing macro definitions.

I raise this as a discussion as this is a little outside my expertise.

**Comment by Jermolene** [comment]
Thanks @AnthonyMuscio 

I think that all of the stylesheets included in the core use macros. As you note, that means that highlighting won't work and I'm afraid that there is no easy way to fix that.

i've no objection to including the highlight plugin on tw.com

**Reply by AnthonyMuscio** [reply]
Background we agree on. If we change a core stylesheet to a tiddler type of stylesheet the css highlighting seems to work quite well, however the macros do not.

Would it not be possible to simply allow the css hightlight to be applied on tiddlers tagged as a stylesheet in addition to those with the css type? This is after all just a way to view a tiddler.

I understand that the highlight makes use of the type as a mime type but it seems that if we can choose to apply the highlight to core stylesheets as well, we gain better self documentation.

In someways this is simply a different type applied to the view template.

**Reply by Jermolene** [reply]
Hi @AnthonyMuscio

> Would it not be possible to simply allow the css hightlight to be applied on tiddlers tagged as a stylesheet in addition to those with the css type? 

I think that could be done by modifying `$:/core/ui/ViewTemplate/body/rendered-plain-text` to contain:

```
<$codeblock code={{!!text}} language="css"/>
```

But I don't think the core should use CSS highlighting for wikitext stylesheet tiddlers because the highlighting will not work reliably.

**Reply by joshuafontany** [reply]
@Jermolene @AnthonyMuscio Would it be possible to have a transcluded view of the spreadsheets w/ highlighting & a ui link to open the tiddler in the story river for viewing/editing the wikitext code? Isn't there a space in $:/ControlPanel? I'll poke around this weekend.

**Reply by Jermolene** [reply]
> @Jermolene @AnthonyMuscio Would it be possible to have a transcluded view of the spreadsheets w/ highlighting & a ui link to open the tiddler in the story river for viewing/editing the wikitext code? Isn't there a space in $:/ControlPanel? I'll poke around this weekend.

I think we have that already in the control panel. This panel displays the rendered stylesheets as seen by the browser. And now in v5.2.3 there is syntax highlighting:

<img width="1170" alt="image" src="https://user-images.githubusercontent.com/174761/181903214-8d08606b-b6db-412e-b940-21e479426842.png">

The presence of that panel is perhaps part of the case for not rendering individual stylesheet tiddlers in the same way.

**Comment by AnthonyMuscio** [comment]
If the highlight plugin is included in the tiddlywiki.com and is displays as on the prerelease this is the desired outcome of the original post.

If @joshuafontany @Jermolene don't have any outstanding issues I / we can close this discussion.

Thanks gentlemen

**Reply by Jermolene** [reply]
Thanks @AnthonyMuscio – this is a discussion thread, so I think instead of closing it there's a way for you to mark one of the posts as being the answer.

**Reply by AnthonyMuscio** [reply]
I cant seem to find where to mark it as an answer.
